packages/flutter_tools/test/integration.shard/isolated/native_assets_test.dart

```diff
@@ -686,6 +692,115 @@ import '${packageName}_bindings_generated.dart' as bindings;
   await dartFile.writeAsString(dartFileNew2);
 }
 
+/// Adds a native library to be built by the builder and link it into the
+/// main library.
+Future<void> addLinkedNativeLibrary(Directory packageDirectory) async {
+  // Add linked library source file.
+  final Directory srcDirectory = packageDirectory.childDirectory('src');
+  final File linkedLibrarySourceFile = srcDirectory.childFile('linked.c');
+  await linkedLibrarySourceFile.writeAsString('''
+#if _WIN32
+#define FFI_PLUGIN_EXPORT __declspec(dllexport)
+#else
+#define FFI_PLUGIN_EXPORT
+#endif
+
+FFI_PLUGIN_EXPORT void linked() {}
+''');
+
+  // Update main library to include call to linked library.
+  final File mainLibrarySourceFile = srcDirectory.childFile('$packageName.c');
+  String mainLibrarySource = await mainLibrarySourceFile.readAsString();
+  mainLibrarySource = mainLibrarySource.replaceFirst(
+    '#include "$packageName.h"',
+'''
+#include "$packageName.h"
+
+extern void linked();
+''',
+  );
+  mainLibrarySource = mainLibrarySource.replaceFirst(
+    'sum(intptr_t a, intptr_t b) {',
+    '''
+sum(intptr_t a, intptr_t b) {
+  linked();
```

Comment:
Nit: It would be slightly cleaner if the return value would come from the linked method. (Though, this will also definitely let the runtime fail with dynamic linking error if it can't find the dylib.)